rubix-core/src/main/java/com/qubole/rubix/core/CachedReadRequestChain.java

```diff
@@ -176,25 +176,12 @@ public void cancel()
 
   private void invalidateMetadata()
   {
-    RetryingBookkeeperClient client = null;
-    try {
-      client = factory.createBookKeeperClient(conf);
+    try (RetryingPooledBookkeeperClient client = factory.createBookKeeperClient(conf)) {
       client.invalidateFileMetadata(remotePath);
     }
     catch (Exception e) {
```

Comment:
This is handled, the client close() just does a put now so shouldnt be throwing any exceptions

Reply:
But, here `invalidateFileMetadata` throws an exception, that'ss why I didn't remove it.

Reply:
Yes, never mind this. I assumed that client.close() can also throw an exception which would be handled by this catch and the message for it would be wrong as invalidateFileMetadata did pass correctly. As mentioned later, it is handled so this not a issue.